src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -357,13 +357,15 @@ def generate_config(
         alertmanager_sd_url = f'{srv_end_point}service=alertmanager' if alertmgr_cnt > 0 else None
         haproxy_sd_url = f'{srv_end_point}service=haproxy' if haproxy_cnt > 0 else None
         mgr_prometheus_sd_url = f'{srv_end_point}service=mgr-prometheus'  # always included
+        ceph_exporter_sd_url = f'{srv_end_point}service=ceph-exporter'  # always included
```

Comment:
Are we sure that this service is always defined/included? shouldn't we include it only if node-exporter daemons are present?

Reply:
No, ceph-exporter is an independent ceph daemon that will run to fetch perf counters of each daemon and expose them as prometheus metrics!

Reply:
> No, ceph-exporter is an independent ceph daemon that will run to fetch perf counters of each daemon and expose them as prometheus metrics!

Understood, the question then is would this daemon be **always running**?

Reply:
Yes it should; it's similar to mgr-prometheus exporter only difference is one is exposing daemon perf counters and other cluster-wide metrics + remaining metrics for different components

Reply:
> Yes it should; it's similar to mgr-prometheus exporter only difference is one is exposing daemon perf counters(which mgr-prometheus was doing before) and other cluster-wide metrics + remaining metrics for different components

Great, thanks for the clarification :+1: